frontend/src/routes/dashboard/admin/index.tsx

```diff
@@ -1,12 +1,121 @@
-import { component$ } from '@builder.io/qwik';
+import { component$, useResource$, Resource, useVisibleTask$, useStore, $ } from '@builder.io/qwik';
 import type { DocumentHead } from '@builder.io/qwik-city';
+import { ServerPaginatedDataTable } from '~/components/table/ServerPaginatedDataTable';
+import { UserCtx } from '~/routes/layout';
+import { authorizedFetch } from '~/shared/auth.service';
+import { fetchMockUsers } from '~/mockdata/useMockFns';
+import { PaginatedRows, PaginationParams } from '~/types/paginated';
 
 export default component$(() => {
+  const firstLoading = useStore({ value: true });
+  const rowsCache = useStore({
+    limit: 10,
+    startIdx: 0,
+    endIdx: 10,
+    rows: [],
+    sort: 'asc',
+    sortColumn: 'id',
+    filter: '',
+    total: 0,
+  });
+
+  const fetchRows = $(async ({ limit, page, filter, sort, sortColumn }: PaginationParams) => {
+    const startIdx = page * limit;
+    const endIdx = startIdx + limit;
+
+    const limitOnServer = 50;
+
+    const isPageInRange = startIdx >= rowsCache.startIdx && endIdx <= rowsCache.endIdx;
+    const isPageInLocalCache = isPageInRange && rowsCache.rows.length === rowsCache.total;
+
+    console.log('trying to fetch cached', {
```

Comment:
Consider remove the logs.

Reply:
Im commenting them out....will remove the comments once better tested